schema/schema.cc

```diff
@@ -838,7 +839,25 @@ std::ostream& schema::describe(replica::database& db, std::ostream& os, bool wit
                 if (n++ != 0) {
                     os << ", ";
                 }
-                os << pk.name_as_cql_string();
+
+                if (is_local) {
+                    // local indexes don't support collection types, so always use just column's name
+                    os << pk.name_as_cql_string();
+                } else {
+                    if (pk.type->is_collection() && !pk.type->is_multi_cell()) {
+                        // Frozen collections always requires full() target while creating an index.
+                        // But the full() target is dropped while saving index information to the table.
+                        // So it needs to be recreated here.
+                        os << "full(" << pk.name_as_cql_string() << ")";
+                    } else {
+                        auto base_schema = db.find_schema(view_info()->base_id());
+                        auto index_metadata = base_schema->all_indices().at(secondary_index::index_name_from_table_name(cf_name()));
+                        auto target = index_metadata.options().at(cql3::statements::index_target::target_option_name);
+                        auto index_target_name = secondary_index::target_parser::get_target_column_name_from_string(target);
```

Comment:
As discussed above, you need to undo here the mess that `create_index_statement::rewrite_target_for_collection() did. The only bad thing it did (I think) is to add the "keys()" when the collection is a set. So if the collection is a set, you need to change it back to values() - which is the only thing that can be indexed in a set.

Other than that, this PR looks good!

I didn't like this "target rewrite" thing when I reviewed the collection indexing code, and now I remember why :-(